src/test/java/io/vertx/core/parsetools/FakeStream.java

```diff
@@ -89,4 +93,12 @@ void fail(Throwable err) {
   void end() {
     endHandler.handle(null);
   }
+
+  public int pauseCount() {
```

Comment:
make both synchronized and I think it will be fine. (this class can be used in multithreaded tests)

Reply:
in such case, instead of int - AtomicInteger should be provided, because 
a. There is race during value incrementation
b. there is no happens-before relationship between adding and reading.

Reply:
I think it's fine, we are not trying to remove race conditions so we don't need happens-before, just ensure safe publication of the value, these values are likely to be used at the end of the test.

Reply:
synchronized keyword do not ensure safe publication, this would be done with volatile on keyword - or do I miss some part of JMM ?

Reply:
It does when the value was stored under synchronization, so we just need to ensure that we use the same monitor and synchronize when we store the value.

Reply:
ok, that makes sense :)

Reply:
@vietj where is corresponding synchronize on writing data (on the same monitor (so instance of FakeStream)) ?

Reply:
I added **volatile** to both **pauseCount** and **resumeCount** - this for sure will ensure visibility correctness between any thread reading and writing (of course do not prevent against races). I could not find example that uses FakeStream in multithreaded environment, so, I could not verify if synchronization keyword could help.
Is that ok ?

Reply:
@KowalczykBartek it is used in rx-java module I haven't checked whether tests are multi-threaded though but there are chances there are.

Reply:
in rx-java module I see that 
```
io.vertx.test.fakestream.Fakestream
```
and that is true, that this class have synchronization in place. But we are talking about 
```
io.vertx.core.parsetools.FakeStream
```

Reply:
ah you are right, so just leave it as is and in the future we'll try to use a single FakeStream class in vertx test classes